sway-core/src/type_engine.rs

```diff
@@ -66,20 +66,21 @@ pub(crate) trait ToJsonAbi {
 impl ToJsonAbi for TypeId {
     fn generate_json_abi(&self) -> Option<Vec<Property>> {
         match look_up_type_id(*self) {
-            TypeInfo::Struct { fields, .. } => {
-                Some(fields.iter().map(|x| x.generate_json_abi()).collect())
-            }
+            TypeInfo::Array(type_id, _) => Some(vec![Property {
+                name: "__array_element".to_string(),
+                type_field: type_id.json_abi_str(),
+                components: type_id.generate_json_abi(),
+            }]),
             TypeInfo::Enum { variant_types, .. } => Some(
                 variant_types
                     .iter()
                     .map(|x| x.generate_json_abi())
                     .collect(),
             ),
-            TypeInfo::Array(type_id, _) => Some(vec![Property {
-                name: "__array_element".to_string(),
-                type_field: type_id.json_abi_str(),
-                components: type_id.generate_json_abi(),
-            }]),
+            TypeInfo::Struct { fields, .. } => {
+                Some(fields.iter().map(|x| x.generate_json_abi()).collect())
+            }
+            TypeInfo::Tuple(fields) => Some(fields.iter().map(|x| x.generate_json_abi()).collect()),
```

Comment:
This is the only line that was added. All other changes in this file are just for sorting the arms in alphabetical ordering.